packages/libs/contracts-sdk/script/UpdateFacet.sol

```diff
@@ -7,235 +7,299 @@ import {console2} from "forge-std/console2.sol";
-                // Check for a common selector in VincentAppViewFacet - using a selector from the library
-                bytes4 getAppSelector = VincentAppViewFacet.getAppById.selector;
-                isTargetFacet = isSameFacetType(facetAddress, getAppSelector);
-            } else if (compareStrings(facetName, "VincentUserFacet")) {
-                // Check for a common selector in VincentUserFacet - using a selector from the library
-                bytes4 registerUserSelector = VincentUserFacet.permitAppVersion.selector;
-                isTargetFacet = isSameFacetType(facetAddress, registerUserSelector);
-            } else if (compareStrings(facetName, "VincentUserViewFacet")) {
-                // Check for a common selector in VincentUserViewFacet - using a selector from the library
-                bytes4 getUserSelector = VincentUserViewFacet.getAllRegisteredAgentPkps.selector;
-                isTargetFacet = isSameFacetType(facetAddress, getUserSelector);
-            }
-
-            if (isTargetFacet) {
-                // Return the function selectors for this facet
+            if (isFacetOfType(facets[i].facetAddress, facetName)) {
                 return facets[i].functionSelectors;
             }
         }
 
-        // If we get here, the facet wasn't found in the diamond
-        console2.log("Warning: Facet not found in diamond. Will use default selectors.");
-
-        // Fall back to hard-coded selectors from the selector helper functions
-        return getDefaultSelectors(facetToUpdate);
+        revert("Facet has no selectors");
     }
 
-    /**
-     * @dev Checks if a facet address contains a specific function selector
-     * @param facetAddress The facet address to check
-     * @param selector The function selector to check for
-     * @return isSameType Whether the facet contains the selector
-     */
-    function isSameFacetType(address facetAddress, bytes4 selector) internal view returns (bool) {
+    function isFacetOfType(address facetAddr, string memory facetName) internal view returns (bool) {
         IDiamondLoupe diamondLoupe = IDiamondLoupe(diamondAddress);
-        address facetForSelector = diamondLoupe.facetAddress(selector);
-        return facetForSelector == facetAddress;
+        
+        bytes4 testSelector;
+        if (compareStrings(facetName, "VincentAppFacet")) {
+            testSelector = VincentAppFacet.registerApp.selector;
+        } else if (compareStrings(facetName, "VincentAppViewFacet")) {
+            testSelector = VincentAppViewFacet.getAppById.selector;
+        } else if (compareStrings(facetName, "VincentUserFacet")) {
+            testSelector = VincentUserFacet.permitAppVersion.selector;
+        } else if (compareStrings(facetName, "VincentUserViewFacet")) {
+            testSelector = VincentUserViewFacet.getPermittedAppVersionForPkp.selector;
+        } else {
+            return false;
+        }
```

Comment:
This relies on the identifying function’s signature never changing. If the function is renamed, its parameters are changed, or it’s removed entirely, the script won’t be able to identify the facet

A simple fix might be just adding a facet name property onto all the facets, something like:
```
function facetName() external pure returns (string memory);
```